File: trust-args.js
/**
 * Startup argv that skip workspace-trust prompts (folder 授信).
 * Only CLIs with a dedicated trust/skip-trust flag — not tool-permission
 * bypasses like --dangerously-skip-permissions / --yolo.
 * OpenCode / Pi / Claude / Codex: no reliable trust flag → empty.
 */
const TRUST_ARGS = {
  cursor: ["--trust"],
  gemini: ["--skip-trust"],
};

/**
 * @param {string} cliId
 * @returns {string[]}
 */
function trustArgsFor(cliId) {
  const args = TRUST_ARGS[cliId];
  return Array.isArray(args) ? args.slice() : [];
}

/**
 * Trust flags first, then resume/continue argv.
 * @param {string} cliId
 * @param {string[]} [resumeArgs]
 * @returns {string[]}
 */
function launchArgsFor(cliId, resumeArgs) {
  const trust = trustArgsFor(cliId);
  const resume = Array.isArray(resumeArgs) ? resumeArgs.filter(Boolean) : [];
  return trust.concat(resume);
}

module.exports = { TRUST_ARGS, trustArgsFor, launchArgsFor };

File: src-tauri/src/trust_args.rs
//! Startup argv that skip workspace-trust prompts.

use std::collections::HashMap;
use std::sync::LazyLock;

static TRUST_ARGS: LazyLock<HashMap<&'static str, Vec<&'static str>>> = LazyLock::new(|| {
    let mut m = HashMap::new();
    m.insert("cursor", vec!["--trust"]);
    m.insert("gemini", vec!["--skip-trust"]);
    // Kimi Code: YOLO — auto-approve tool calls (Win + Mac). Mutually exclusive with --auto.
    m.insert("kimi", vec!["--yolo"]);
    m
});

pub fn trust_args_for(cli_id: &str) -> Vec<String> {
    TRUST_ARGS
        .get(cli_id)
        .map(|v| v.iter().map(|s| (*s).to_string()).collect())
        .unwrap_or_default()
}

pub fn launch_args_for(cli_id: &str, resume_args: &[String]) -> Vec<String> {
    let mut out = trust_args_for(cli_id);
    out.extend(resume_args.iter().cloned().filter(|s| !s.is_empty()));
    out
}

#[cfg(test)]
mod tests {
    use super::*;

    #[test]
    fn trust_cursor_gemini_kimi() {
        assert_eq!(trust_args_for("cursor"), vec!["--trust".to_string()]);
        assert_eq!(trust_args_for("gemini"), vec!["--skip-trust".to_string()]);
        assert_eq!(trust_args_for("kimi"), vec!["--yolo".to_string()]);
        assert!(trust_args_for("claude").is_empty());
    }

    #[test]
    fn launch_trust_then_resume() {
        let args = launch_args_for("cursor", &["--continue".into()]);
        assert_eq!(args, vec!["--trust".to_string(), "--continue".to_string()]);
    }
}
